terra_layer/style.py

```diff
@@ -0,0 +1,397 @@
+    return 2 * math.sqrt(value * max_size / max_value / math.pi)
+
+
+def circle_boundaries_filter_values(values, max_value, max_size, dmin):
+    if not values or max_value is None or max_size is None:
+        return []
+
+    # array that will hold the filtered values
+    filtered_values = []
+    # add the maximum value
+    filtered_values.append(values[0])
+    # remember the height of the previously added value
+    previous_height = circle_boundaries_value_to_symbol_height(
+        values[0], max_value, max_size
+    )
+    # find the height and value of the smallest acceptable symbol
+    last_height = 0
+    last_value_id = len(values) - 1
+    while last_value_id >= 0:
+        last_height = circle_boundaries_value_to_symbol_height(
+            values[last_value_id], max_value, max_size
+        )
+        if last_height > dmin:
+            break
+        last_value_id -= 1
+
+    # loop over all values that are large enough
+    for limit_id in range(1, last_value_id + 1):
+        v = values[limit_id - 1]
+        # compute the height of the symbol
+        h = circle_boundaries_value_to_symbol_height(v, max_value, max_size)
+        # do not draw the symbol if it is too close to the smallest symbol (but is not the smallest limit itself)
+        if h - last_height < dmin and limit_id != last_value_id:
+            continue
+        # do not draw the symbol if it is too close to the previously drawn symbol
+        if previous_height - h < dmin:
+            continue
+        filtered_values.append(v)
+        # remember the height of the last drawn symbol
+        previous_height = h
+
+    return filtered_values
+
+
+def gen_legend_circle(min, max, size):
+    """
+    Generate a circle legend.
+    """
+    candidates = circle_boundaries_candidate(min, max)
+    candidates = [max] + candidates + [min]
+    boundaries = circle_boundaries_filter_values(candidates, min, max, size / 20)[::-1]
+    return [
+        {"radius": b * size / max, "label": f"{b}", "shape": "circle"}
+        for b in boundaries
+    ]
+
+
+def gen_layer_fill(fill_color=DEFAULT_FILL_COLOR, stroke_color=DEFAULT_STROKE_COLOR):
+    """
+    Build a Mapbox GL Style layer for pylygon fill.
+    """
+    return {
+        "type": "fill",
+        "paint": {"fill-color": fill_color, "fill-outline-color": stroke_color},
+    }
+
+
+def gen_layer_circle(
+    radius, fill_color=DEFAULT_FILL_COLOR, stroke_color=DEFAULT_STROKE_COLOR
+):
+    """
+    Build a Mapbox GL Style layer for circle.
+    """
+    return {
+        "type": "circle",
+        "paint": {"circle-radius": radius, "circle-stroke-color": stroke_color},
+    }
+
+
+def generate_style_from_wizard(layer, config):
+    """
+    Return a Mapbox GL Style and a Legend from a wizard setting.
+    """
+    geo_layer = layer.source.get_layer()
+    symbology = config["symbology"]
+    property = config["property"]
+
+    if symbology == "graduated":
```

Comment:
Can we use `choropleth` instead of graduated here ? (See https://en.wikipedia.org/wiki/Choropleth_map)

Reply:
Yes, but "choropleth" is the resulting map, not the computation method.